icu4c/source/i18n/number_grouping.cpp

```diff
@@ -64,6 +64,13 @@ Grouper Grouper::forProperties(const DecimalFormatProperties& properties) {
 }
 
 void Grouper::setLocaleData(const impl::ParsedPatternInfo &patternInfo, const Locale& locale) {
+    if (fMinGrouping == -2) {
```

Comment:
I have hard time to understand how the code jump from DecimalFormat::setMinimumGroupingDigits to here. Could you explain the relationship between  DecimalFormat and Grouper? 
Is there a design doc I can see to understand the internal relationship between these classes/objects?

Reply:
Grouper is for NumberFormatter, but also indirectly for DecimalFormat.  The DecimalFormat properties gets turned into NumberFormatter properties via number_mapper.cpp.

Reply:
Note that -2 and -3 are already supported internally in NumberFormatter.  I'm just exposing them to DecimalFormat.

I had to move these 7 lines of code to the top of the function in order for them to get evaluated even when fGrouping1 and fGrouping2 are set, as is the case in the DecimalFormat code path.